core/src/main/java/tc/oc/pgm/filters/dynamic/Filterables.java

```diff
@@ -28,4 +31,12 @@ public static Class<? extends Filterable<?>> scope(Filter filter) {
             + filter.getQueryType().getSimpleName()
             + " does not have a filterable scope");
   }
+
+  /** Extract the most specific filterable possible from the given query */
+  public static Filterable<?> extractFilterable(MatchQuery query) {
```

Comment:
this method could probably be in MatchQuery interface, and the Filterables.scope could probably just be in the Filter interface